javascript/javascript2/week3/class_task/script.js

```diff
@@ -0,0 +1,60 @@
+const button1 = document.getElementById("button1");
+const button2 = document.getElementById("button2");
+const counter = document.getElementById("counter");
+button1.addEventListener("click", increment);
+button2.addEventListener("click", increment);
+
+function increment(e) {
+  if (counter.innerText) {
+    counter.innerText = Number(counter.innerText) + 1;
+  } else {
+    counter.innerText = 0;
+  }
+}
+
+const delayClicker = document.getElementById("delay-clicker");
+
+delayClicker.addEventListener("click", () =>
+  setTimeout(() => console.log("Logged in a 3 seconds"), 3000)
+);
+
+document.addEventListener("DOMContentLoaded", (event) => {
+  console.log("DOM fully loaded and parsed");
+});
+window.addEventListener("pointerup", (e) => {
+  console.log(e.x, e.y);
+});
+
+const coordinates = [];
+const trackMouse = (e) => coordinates.push({ x: e.x, y: e.y });
+window.addEventListener("pointermove", trackMouse);
+
+const intervalId = setInterval(calculateAverage, 5000);
+
+function calculateAverage() {
+  const averageCoordinates = {};
+  averageCoordinates.averageX =
+    coordinates.reduce((acc, curr) => {
+      acc.x = acc.x + curr.x;
```

Comment:
Can simplify this with `coordinates.reduce((acc, curr) => acc.x + curr.x).x`

Reply:
I tried to do coordinates.reduce((acc, curr) => acc.x + curr.x); initially, but after the first iteration acc becomes a number, not object. So on a second iteration the call to acc.x returns undefined and the whole function fails. Hense, I had to return the object each time the reduce iterates on the array.

Reply:
My bad, I meant `coordinates.reduce((acc, curr) => acc + curr.x, 0)`. You can specify the starting value of acc in the second argument of the reduce function so it does not need to be the first value of the array. You don't need it to be an object and do .x afterwards. Alternatively, you can have acc it be an object and do both x and y sums together.

Reply:
rewrote it to retur the object with the both coordinates summed. Then just divide both properties to the respective length.